x-pack/plugins/apm/public/components/shared/tutorial/config_agent/index.tsx

```diff
@@ -0,0 +1,141 @@
+import { i18n } from '@kbn/i18n';
+import { HttpStart } from 'kibana/public';
+import React, { useEffect, useMemo, useState } from 'react';
+import styled from 'styled-components';
+import { APIReturnType } from '../../../../services/rest/createCallApmApi';
+import { CopyCommands } from '../copy_commands';
+import { PolicySelectorOption, PolicySelector } from './policy_selector';
+import { getCommands } from './commands/get_commands';
+import { getPolicyOptions } from './get_policy_options';
+
+export type APIResponseType = APIReturnType<'GET /api/apm/fleet/agents'>;
+
+const CentralizedContainer = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`;
+
+const MANAGE_FLEET_POLICIES_LABEL = i18n.translate(
+  'xpack.apm.tutorial.agent_config.manageFleetPolicies',
+  { defaultMessage: 'Manage fleet policies' }
+);
+
+const GET_STARTED_WITH_FLEET_LABEL = i18n.translate(
+  'xpack.apm.tutorial.agent_config.getStartedWithFleet',
+  { defaultMessage: 'Get started with fleet' }
+);
+
+interface Props {
+  variantId: string;
+  http: HttpStart;
+  basePath: string;
+  isCloudEnabled: boolean;
+}
+
+function TutorialAgentSecretTokenSelector({
+  variantId,
+  http,
+  basePath,
+  isCloudEnabled,
+}: Props) {
+  const [data, setData] = useState<APIResponseType>({
+    agents: [],
+    cloudStandaloneSetup: undefined,
+  });
+  const [isLoading, setIsLoading] = useState(true);
+  const [selectedOption, setSelectedOption] = useState<PolicySelectorOption>();
+
+  useEffect(() => {
+    async function fetchData() {
+      setIsLoading(true);
+      try {
+        const response = await http.get('/api/apm/fleet/agents');
+        if (response) {
+          setData(response as APIResponseType);
+        }
+      } catch (e) {
+        console.error('Error while fetching fleet agents.', e);
+      }
+    }
+    fetchData();
+  }, [http]);
+
+  // Depending the environment running (onPrem/Cloud) different values must be available and automatically selected
+  const options = useMemo(() => {
+    const { availableOptions, defaultSelectedOption } = getPolicyOptions({
+      isCloudEnabled,
+      data,
+    });
+    setSelectedOption(defaultSelectedOption);
+    setIsLoading(false);
+    return availableOptions;
+  }, [data, isCloudEnabled]);
+
+  if (isLoading) {
+    return (
+      <CentralizedContainer data-test-subj="loading">
+        <EuiLoadingSpinner />
+      </CentralizedContainer>
+    );
+  }
+
+  const commands = getCommands({
+    variantId,
+    policyDetails: {
+      apmServerUrl: selectedOption?.apmServerUrl,
+      secretToken: selectedOption?.secretToken,
+    },
+  });
+
+  const hasFleetAgents = !!data.agents.length;
+  const fleetLink = hasFleetAgents
+    ? {
+        label: MANAGE_FLEET_POLICIES_LABEL,
+        href: `${basePath}/app/fleet#/policies`,
+      }
+    : {
+        label: GET_STARTED_WITH_FLEET_LABEL,
+        href: `${basePath}/app/integrations#/detail/apm-0.2.0/overview`,
```

Comment:
@ogupte did you figure how to retrieve the apm version, I also need to link to the correct apm integration page.